controller/server_controller.py

```diff
@@ -125,31 +123,45 @@ def cull_idle_jupyter_servers(body, name, namespace, logger, **kwargs):
     js_server_status = get_js_server_status(body)
     custom_resource_api = get_api(config.api_version, config.custom_resource_name)
     idle_seconds = int(body["status"].get("idleSeconds", 0))
+    now = pytz.UTC.localize(datetime.utcnow())
+    last_activity = now if js_server_status is None else js_server_status.get("last_activity", now)
```

Comment:
@ableuler I essentially tried to follow the what the [jupyterhub culler does](https://github.com/jupyterhub/jupyterhub-idle-culler/blob/main/jupyterhub_idle_culler/__init__.py#L213-L222). According to the code there older versions (jupyterhub <0.9) can be missing this field.

Reply:
Also when the jupyter server is not still fully started then the whole endpoint is unresponsive. In that case the value of `js_server_status` is `None`. I wanted to avoid mostly this scenario.

Reply:
See here: https://github.com/SwissDataScienceCenter/amalthea/blob/fix-culling-and-probe-endpoint/controller/culling.py#L74. The function that gets the jupyter server status can respond with `None` in several cases. So it feels like we are setting ourselves up to just have a bunch of unhandled exceptions if I remove all logic and just replace it with `js_server_status.["last_activity"].

Reply:
Therefore I think that we should not really change this.